src/engine/effects/engineeffectchain.cpp

```diff
@@ -128,50 +128,42 @@ bool EngineEffectChain::processEffectsRequest(const EffectsRequest& message,
     return true;
 }
 
-bool EngineEffectChain::enableForGroup(const QString& group) {
-    GroupStatus& status = getGroupStatus(group);
+bool EngineEffectChain::enableForChannel(const ChannelHandle& group) {
+    ChannelStatus& status = getChannelStatus(group);
     if (status.enable_state != EffectProcessor::ENABLED) {
         status.enable_state = EffectProcessor::ENABLING;
     }
     return true;
 }
 
-bool EngineEffectChain::disableForGroup(const QString& group) {
-    GroupStatus& status = getGroupStatus(group);
+bool EngineEffectChain::disableForChannel(const ChannelHandle& group) {
+    ChannelStatus& status = getChannelStatus(group);
     if (status.enable_state != EffectProcessor::DISABLED) {
         status.enable_state = EffectProcessor::DISABLING;
     }
     return true;
 }
 
-EngineEffectChain::GroupStatus& EngineEffectChain::getGroupStatus(const QString& group) {
-    for (QLinkedList<GroupStatus>::iterator it = m_groupStatus.begin(),
-                 end = m_groupStatus.end(); it != end; ++it) {
-        GroupStatus& status = *it;
-        if (status.group == group) {
-            return status;
-        }
-    }
-    GroupStatus status(group);
-    m_groupStatus.append(status);
-    return m_groupStatus.last();
+EngineEffectChain::ChannelStatus& EngineEffectChain::getChannelStatus(
+        const ChannelHandle& group) {
+    return m_channelStatus[group];
 }
 
-void EngineEffectChain::process(const QString& group,
+void EngineEffectChain::process(const ChannelHandle& group,
```

Comment:
Indeed, I think this actually makes the code simpler. Right now there are two "groups" in the effect system -- the group of the processing code (e.g. ```[EffectRackN_EffectUnitM]```) and the group of the channel the effect unit is processing (e.g. ```[Channel]```). It's easy to confuse the two since the code said things like "registeredGroups" when it meant channel groups.